test-data/unit/check-newsemanal.test

```diff
@@ -2200,3 +2201,28 @@ x = 1
 # We want to check that the first definition creates the variable.
 def x() -> None: ...  # E: Name 'x' already defined on line 1
 y = 2
+
+[case testNewAnalyzerImportStarSpecialCase]
+import unittest
+[file unittest/__init__.pyi]
+from unittest.suite import *
+
+def load_tests() -> TestSuite: ...
+[file unittest/suite.pyi]
+from typing import Union # Iterable not imported
+import unittest.case
+
+_TestType = Union[unittest.case.TestCase]
```

Comment:
Maybe also add a test with the same alias, but without the union?

Reply:
I'm not sure if it will be provide useful coverage. The union may not be necessary, but it's here because this was derived from the real stubs for `unittest`, and they happen to contain a union :-)